pumps/resurface_test.go

```diff
@@ -0,0 +1,333 @@
+		assert.Contains(t, message, "[\"response_header:access-control-allow-credentials\",\"true\"]")
+		assert.Contains(t, message, "[\"response_header:access-control-allow-origin\",\"*\"]")
+		assert.Contains(t, message, "[\"response_header:content-type\",\"application/json\"]")
+		assert.Contains(t, message, "[\"response_header:content-type\",\"application/json\"]")
+		assert.Contains(t, message, "[\"response_body")
+		assert.Contains(t, message, "Yours Truly")
+	}
+
+	err = pmp.WriteData(context.TODO(), []interface{}{
+		analytics.AnalyticsRecord{
+			Host:         MOCK_HOST,
+			Method:       "PUT",
+			ResponseCode: 404,
+			RawRequest:   "bm90IHZhbGlkCg==",
+			RawResponse:  "bm90IHZhbGlkCg==",
+			TimeStamp:    time.Now(),
+		},
+	})
+	if err != nil {
+		t.Fatal(pmp.GetName()+"couldn't write records with err:", err)
+	}
+	queue = pmp.logger.Queue()
+	assert.Equal(t, len(recs)+1, len(queue))
+
+	message := queue[len(queue)-1]
+	assert.Contains(t, message, "[\"request_url\",\"http://"+MOCK_HOST)
+	assert.NotContains(t, message, "[\"request_url\",\"http://localhost:8080/get\"]")
+	assert.NotContains(t, message, "[\"request_method\",\"PUT\"]")
+	assert.NotContains(t, message, "[\"request_method\",\"GET\"]")
+	assert.NotContains(t, message, "request_header")
+
+	assert.Contains(t, message, "response_code\",\"404")
+	assert.NotContains(t, message, "response_code\",\"200")
+	assert.NotContains(t, message, "response_header")
+	assert.NotContains(t, message, "response_body")
+	assert.NotContains(t, message, "Yours Truly")
+
+}
+
+func TestResurfaceWriteCustomFields(t *testing.T) {
+	pmp, _ := SetUp(t, "", make([]string, 0), "include debug")
+
+	recs := []interface{}{
+		analytics.AnalyticsRecord{
+			APIID:        "my-api-123",
+			OrgID:        "my-org-abc",
+			Host:         "testone",
+			Method:       "GET",
+			ResponseCode: 200,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			APIID:        " hello  ",
+			OrgID:        "  world",
+			Host:         "testtwo",
+			Method:       "POST",
+			ResponseCode: 200,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			APIID:        "727dad853a8a45f64ab981154d1ffdad",
+			APIKey:       "an-uhashed-key",
+			APIName:      "Foo API",
+			APIVersion:   "0.1.0-b",
+			OauthID:      "my-oauth-client-id",
+			OrgID:        "my-org-abc",
+			Host:         "test-3",
+			Method:       "GET",
+			ResponseCode: 500,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			APIID:       "",
+			OrgID:       "",
+			Host:        "test-four",
+			Method:      "GET",
+			RawRequest:  rawReq,
+			RawResponse: rawResp,
+			TimeStamp:   time.Now(),
+		},
+	}
+
+	err := pmp.WriteData(context.TODO(), recs)
+	if err != nil {
+		t.Fatal(pmp.GetName()+"couldn't write records with err:", err)
```

Comment:
we can use `assert.NotNil` here

Reply:
Thank you for pointing this out, I just modified it.